pkg/virt-operator/resource/apply/delete_test.go

```diff
@@ -169,6 +175,92 @@ var _ = Describe("Deletion", func() {
 			Expect(needRemoved).To(BeEmpty())
 		})
 	})
+
+	Context("Node label deletion", func() {
+		var clientset *fake.Clientset
+		var kubevirtClient *kubecli.MockKubevirtClient
+		var ctrl *gomock.Controller
+
+		BeforeEach(func() {
+			ctrl = gomock.NewController(GinkgoT())
+			clientset = fake.NewSimpleClientset()
+			kubevirtClient = kubecli.NewMockKubevirtClient(ctrl)
+			kubevirtClient.EXPECT().CoreV1().Return(clientset.CoreV1()).AnyTimes()
+		})
+
+		AfterEach(func() {
+			ctrl.Finish()
+		})
+
+		It("should delete kubevirt labels from nodes", func() {
+			nodes := &k8sv1.NodeList{
+				Items: []k8sv1.Node{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "node1",
+							Labels: map[string]string{
+								"kubevirt.io/schedulable": "true",
+								"kubevirt.io/some-label":  "value",
+								"other-label":             "value",
+							},
+						},
+					},
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "node2",
+							Labels: map[string]string{
+								"kubevirt.io/schedulable":   "true",
+								"kubevirt.io/another-label": "value",
+							},
+						},
+					},
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:   "node3",
+							Labels: map[string]string{},
+						},
+					},
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "node4",
+							Labels: map[string]string{
+								"kubevirt.io/another-label": "value",
+							},
+						},
+					},
+				},
+			}
+
+			_, err := clientset.CoreV1().Nodes().Create(context.Background(), &nodes.Items[0], metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			_, err = clientset.CoreV1().Nodes().Create(context.Background(), &nodes.Items[1], metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			_, err = clientset.CoreV1().Nodes().Create(context.Background(), &nodes.Items[2], metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			_, err = clientset.CoreV1().Nodes().Create(context.Background(), &nodes.Items[3], metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			err = deleteKubeVirtLabelsFromNodes(kubevirtClient)
+			Expect(err).ToNot(HaveOccurred())
+
+			updatedNode1, err := clientset.CoreV1().Nodes().Get(context.Background(), "node1", metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(updatedNode1.Labels).ToNot(HaveKey("kubevirt.io/some-label"))
+			Expect(updatedNode1.Labels).To(HaveKey("other-label"))
+
+			updatedNode2, err := clientset.CoreV1().Nodes().Get(context.Background(), "node2", metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(updatedNode2.Labels).ToNot(HaveKey("kubevirt.io/another-label"))
+
+			updatedNode3, err := clientset.CoreV1().Nodes().Get(context.Background(), "node3", metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(updatedNode3.Labels).To(BeEmpty())
```

Comment:
It will be good to check that no patch has been performed because it was filtered out.
Maybe adding something like this:
```
			Expect(clientset.Actions()).To(WithTransform(func(actions []testing.Action) []testing.Action {
				var node3patchActions []testing.Action
				for _, action := range actions {
					if action.GetVerb() == "patch" &&
						action.GetResource().Resource == "nodes" {
						patchAction := action.(testing.PatchAction)
						if patchAction.GetName() == "node3" {
							node3patchActions = append(node3patchActions, action)
						}
					}
				}
				return node3patchActions
			}, BeEmpty()))
```

Reply:
@fossedihelm i made all changes based on your review.